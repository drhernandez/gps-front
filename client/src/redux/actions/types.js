const TOGGLE_SIDEBAR = 'TOGGLE_SIDEBAR';
const SET_USER_INFO = 'SET_USER_INFO';
const REMOVE_USER_INFO = 'REMOVE_USER_INFO';

export {
  TOGGLE_SIDEBAR,
  SET_USER_INFO,
  REMOVE_USER_INFO
};