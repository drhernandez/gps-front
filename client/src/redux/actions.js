const TOGGLE_SIDEBAR = 'TOGGLE_SIDEBAR';

function toogleSidebarAction() {
  return {
    type: TOGGLE_SIDEBAR
  }
}

const types = {
  TOGGLE_SIDEBAR
};

const actions = {
  toogleSidebarAction
}

export { types, actions };