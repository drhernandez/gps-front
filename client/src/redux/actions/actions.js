import { TOGGLE_SIDEBAR } from './types';

function toogleSidebarAction() {
  return {
    type: TOGGLE_SIDEBAR
  }
}

export {
  toogleSidebarAction
};