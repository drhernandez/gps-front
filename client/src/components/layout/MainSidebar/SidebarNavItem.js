import React from "react";
import PropTypes from "prop-types";
import { connect } from 'react-redux'
import { NavLink as RouteNavLink } from "react-router-dom";
import { NavItem, NavLink } from "shards-react";

import { toogleSidebarAction } from "../../../redux/actions/actions";

const SidebarNavItem = (props) => (
  <NavItem>
    <NavLink tag={RouteNavLink} to={props.item.to} onClick={props.toogleSidebarAction}>
      {props.item.htmlBefore && (
        <div
          className="d-inline-block item-icon-wrapper"
          dangerouslySetInnerHTML={{ __html: props.item.htmlBefore }}
        />
      )}
      {props.item.title && <span>{props.item.title}</span>}
      {props.item.htmlAfter && (
        <div
          className="d-inline-block item-icon-wrapper"
          dangerouslySetInnerHTML={{ __html: props.item.htmlAfter }}
        />
      )}
    </NavLink>
  </NavItem>
);

SidebarNavItem.propTypes = {
  /**
   * The item object.
   */
  item: PropTypes.object
};

const mapDispatchToProps = dispatch => {
  return {
    toogleSidebarAction: () => dispatch(toogleSidebarAction())
  }
}

export default connect(null, mapDispatchToProps)(SidebarNavItem);
