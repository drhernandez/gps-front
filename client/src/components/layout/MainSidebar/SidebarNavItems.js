import React from "react";
import { connect } from 'react-redux'
import { Nav } from "shards-react";

import SidebarNavItem from "./SidebarNavItem";

const mapStateToProps = state => {
  return { navItems: state.navItems };
};

class SidebarNavItems extends React.Component {

  render() {
    // const { navItems: items } = this.state;
    return (
      <div className="nav-wrapper">
        <Nav className="nav--no-borders flex-column">
          {this.props.navItems.map((item, idx) => (
            <SidebarNavItem key={idx} item={item} />
          ))}
        </Nav>
      </div>
    )
  }
}

export default connect(mapStateToProps, null)(SidebarNavItems);
