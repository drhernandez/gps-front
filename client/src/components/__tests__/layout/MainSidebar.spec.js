import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { Router } from "react-router-dom"
import { createMemoryHistory } from 'history'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
const mockStore = configureStore([])

import { toogleSidebarAction } from "../../../redux/actions/actions"
import MainSidebar from '../../layout/MainSidebar/MainSidebar'
import SidebarMainNavbar from '../../layout/MainSidebar/SidebarMainNavbar'
import SidebarNavItem from '../../layout/MainSidebar/SidebarNavItem'
import SidebarNavItems from '../../layout/MainSidebar/SidebarNavItems'
import getItems from '../../../data/sidebar-nav-items'

let history
let store
describe('MainSidebar components', () => {
  beforeEach(() => {
    jest.resetModules()
    history = createMemoryHistory()
    store = mockStore({
      menuVisible: true,
      navItems: getItems(),
      userInfo: {
        role: {
          name: 'ADMIN'
        }
      }
    })
    store.dispatch = jest.fn()
  })

  it('MainSidebar snapshot', () => {
    const component = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <MainSidebar />
        </Provider>
      </Router>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('SidebarMainNavbar snapshot', () => {
    const component = renderer.create(
        <Provider store={store}>
          <SidebarMainNavbar />
        </Provider>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('SidebarMainNavbar should dispatch toogleSidebarAction', () => {
    const wrapper = mount(
        <Provider store={store}>
          <SidebarMainNavbar />
        </Provider>
    )

    wrapper.find('a').at(1).simulate('click')

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(toogleSidebarAction());
  })

  it('SidebarNavItem snapshot', () => {
    const props = {
      item: {
        title: "Mi ubicaci√≥n",
        to: "/home",
        htmlBefore: '<i class="material-icons">my_location</i>',
        htmlAfter: ""
      }
    }

    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <SidebarNavItem {...props} />
        </Router>
      </Provider>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('SidebarNavItems snapshot', () => {
    const component = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <SidebarNavItems />
        </Provider>
      </Router>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('SidebarNavItems should have 3 nav items', () => {
    const wrapper = mount(
      <Router history={history}>
        <Provider store={store}>
          <SidebarNavItems />
        </Provider>
      </Router>
    )

    expect(wrapper.find('SidebarNavItem')).toHaveLength(3)
  })
})