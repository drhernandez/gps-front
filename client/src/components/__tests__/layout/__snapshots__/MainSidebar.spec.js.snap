// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSidebar components MainSidebar snapshot 1`] = `
<aside
  className="main-sidebar px-0 col-12 open col-md-3 col-lg-2"
>
  <div
    className="main-navbar"
  >
    <nav
      className="align-items-stretch bg-white flex-md-nowrap border-bottom p-0 navbar navbar-light"
    >
      <a
        className="w-100 mr-0 navbar-brand"
        href="#"
        style={
          Object {
            "lineHeight": "25px",
          }
        }
      >
        <div
          className="d-table m-auto"
        >
          <img
            alt="Logo"
            className="d-inline-block align-top mr-1"
            id="main-logo"
            src={Object {}}
            style={
              Object {
                "maxWidth": "25px",
              }
            }
          />
          <span
            className="d-none d-md-inline ml-1"
          >
            S.G.S.
          </span>
        </div>
      </a>
      <a
        className="toggle-sidebar d-sm-inline d-md-none d-lg-none"
        onClick={[Function]}
      >
        <i
          className="material-icons"
          id="hoa"
        >
          
        </i>
      </a>
    </nav>
  </div>
  <div
    className="nav-wrapper"
  >
    <ul
      className="nav--no-borders flex-column nav"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/new-client"
          onClick={[Function]}
        >
          <div
            className="d-inline-block item-icon-wrapper"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<i class=\\"material-icons\\">person_add</i>",
              }
            }
          />
          <span>
            Alta de cliente
          </span>
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/clients"
          onClick={[Function]}
        >
          <div
            className="d-inline-block item-icon-wrapper"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<i class=\\"material-icons\\">build</i>",
              }
            }
          />
          <span>
            Administrar clientes
          </span>
          
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/tracking"
          onClick={[Function]}
        >
          <div
            className="d-inline-block item-icon-wrapper"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<i class=\\"material-icons\\">my_location</i>",
              }
            }
          />
          <span>
            Tracking de clientes
          </span>
          
        </a>
      </li>
    </ul>
  </div>
</aside>
`;

exports[`MainSidebar components SidebarMainNavbar snapshot 1`] = `
<div
  className="main-navbar"
>
  <nav
    className="align-items-stretch bg-white flex-md-nowrap border-bottom p-0 navbar navbar-light"
  >
    <a
      className="w-100 mr-0 navbar-brand"
      href="#"
      style={
        Object {
          "lineHeight": "25px",
        }
      }
    >
      <div
        className="d-table m-auto"
      >
        <img
          alt="Logo"
          className="d-inline-block align-top mr-1"
          id="main-logo"
          src={Object {}}
          style={
            Object {
              "maxWidth": "25px",
            }
          }
        />
        <span
          className="d-none d-md-inline ml-1"
        >
          S.G.S.
        </span>
      </div>
    </a>
    <a
      className="toggle-sidebar d-sm-inline d-md-none d-lg-none"
      onClick={[Function]}
    >
      <i
        className="material-icons"
        id="hoa"
      >
        
      </i>
    </a>
  </nav>
</div>
`;

exports[`MainSidebar components SidebarNavItem snapshot 1`] = `
<li
  className="nav-item"
>
  <a
    aria-current={null}
    className="nav-link"
    href="/home"
    onClick={[Function]}
  >
    <div
      className="d-inline-block item-icon-wrapper"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<i class=\\"material-icons\\">my_location</i>",
        }
      }
    />
    <span>
      Mi ubicación
    </span>
    
  </a>
</li>
`;

exports[`MainSidebar components SidebarNavItems snapshot 1`] = `
<div
  className="nav-wrapper"
>
  <ul
    className="nav--no-borders flex-column nav"
  >
    <li
      className="nav-item"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/new-client"
        onClick={[Function]}
      >
        <div
          className="d-inline-block item-icon-wrapper"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<i class=\\"material-icons\\">person_add</i>",
            }
          }
        />
        <span>
          Alta de cliente
        </span>
        
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/clients"
        onClick={[Function]}
      >
        <div
          className="d-inline-block item-icon-wrapper"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<i class=\\"material-icons\\">build</i>",
            }
          }
        />
        <span>
          Administrar clientes
        </span>
        
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/tracking"
        onClick={[Function]}
      >
        <div
          className="d-inline-block item-icon-wrapper"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<i class=\\"material-icons\\">my_location</i>",
            }
          }
        />
        <span>
          Tracking de clientes
        </span>
        
      </a>
    </li>
  </ul>
</div>
`;
